brainglobe_atlasapi/atlas_generation/atlas_scripts/cavefish_atlas.py

```diff
@@ -0,0 +1,242 @@
+
+    # download atlas files
+    utils.check_internet_connection()
+    destination_path = download_dir_path / "atlas_download"
+    utils.retrieve_over_http(ATLAS_FILE_URL, destination_path)
+
+    # unpack the atlas download folder
+    with zipfile.ZipFile(destination_path, "r") as zip:
+        zip.extractall(path=atlas_path)
+
+    destination_path.unlink()
+
+    structures_file = atlas_path / "asty_atlas/SPF2_25_Region_atlas_list.csv"
+    annotations_file = (
+        atlas_path / "asty_atlas/SPF2_regions_SP2c_1iWarp_25.tif"
+    )
+    # reference_cartpt = atlas_path / "SPF2_carpt_ref.tif" #ADDITIONAL REFERENCE
+    reference_file = atlas_path / "asty_atlas/SPF2_terk_ref.tif"
+    meshes_dir_path = atlas_path / "asty_atlas/meshes"
+    try:
+        os.mkdir(meshes_dir_path)
+    except:
+        "mesh folder already exists"
+
+    # cartpt = tifffile.imread(reference_cartpt)
+    # ADDITIONAL_REFERENCES = {"cartpt": cartpt}
+
+    # create dictionaries
+    print("Creating structure tree")
+    with open(structures_file, mode="r", encoding="utf-8-sig") as zfishFile:
+        zfishDictReader = csv.DictReader(zfishFile)
+
+        # empty list to populate with dictionaries
+        hierarchy = []
+
+        # parse through csv file and populate hierarchy list
+        for row in zfishDictReader:
+            hierarchy.append(row)
+
+        hierarchy = hierarchy[:-1]
+    # make string to int and list of int conversions in
+    # 'id', 'structure_id_path', and 'rgb_triplet' key values
+    for i in range(0, len(hierarchy)):
+        hierarchy[i]["id"] = int(hierarchy[i]["id"])
+    for j in range(0, len(hierarchy)):
+        hierarchy[j]["structure_id_path"] = list(
+            map(int, hierarchy[j]["structure_id_path"].split("/"))
+        )
+    for k in range(0, len(hierarchy)):
+        try:
+            hierarchy[k]["rgb_triplet"] = list(
+                map(int, hierarchy[k]["rgb_triplet"].split("/"))
+            )
+        except ValueError:
+            hierarchy[k]["rgb_triplet"] = [255, 255, 255]
+
+    # remove clear label (id 0) from hierarchy.
+    # ITK-Snap uses this to label unlabeled areas,
+    # but this convention interferes with the root mask generation
+    # and is unnecessary for this application
+    hierarchy.remove(hierarchy[1])
+
+    # use tifffile to read annotated file
+    annotated_volume = tifffile.imread(annotations_file)
+    reference_volume = tifffile.imread(reference_file)
+
+    from scipy.ndimage import zoom
+
+    annotated_volume = zoom(annotated_volume, (1, 1, 4), order=0)
+
+    print(f"Saving atlas data at {atlas_path}")
+    tree = get_structures_tree(hierarchy)
+    print(
+        f"Number of brain regions: {tree.size()}, "
+        f"max tree depth: {tree.depth()}"
+    )
+
+    # generate binary mask for mesh creation
+    labels = np.unique(annotated_volume).astype(np.int_)
+    for key, node in tree.nodes.items():
+        if key in labels:
+            is_label = True
```

Comment:
You can just write
`is_label = (key in labels)`

Reply:
Thanks @VascoSch92 - I'd say the existing code is probably easier for novices to read, while your suggestion saves a line of code :shrug: Maybe leaning towards legibility for novice programmers, as the atlas packaging and generation code is an area where we particularly encourage people with a non-computational background to engage?

(Also, this script doesn't work as expected yet, so I'll work on that first :sweat_smile: - that's why this PR is in draft mode)

This is not your first review, and I am excited about anyone getting involved with BrainGlobe - If you'd like to discuss more about coordinating possible contributions, can I suggest you join [our zulip chat](https://brainglobe.zulipchat.com/) ?

Reply:
Hey,

thanks for the invitation ;-) 
I will give a look in chat :-D